src/adapters/MorphoMarketV1Adapter.sol

```diff
@@ -73,19 +81,24 @@ contract MorphoMarketV1Adapter is IMorphoMarketV1Adapter {
     /// @dev Does not log anything because the ids (logged in the parent vault) are enough.
     /// @dev Returns the ids of the allocation and the change in allocation.
     function allocate(bytes memory data, uint256 assets, bytes4, address) external returns (bytes32[] memory, int256) {
-        MarketParams memory marketParams = abi.decode(data, (MarketParams));
+        require(data.length == 0, InvalidData());
         require(msg.sender == parentVault, NotAuthorized());
-        require(marketParams.loanToken == asset, LoanAssetMismatch());
 
-        if (assets > 0) IMorpho(morpho).supply(marketParams, assets, 0, address(this), hex"");
+        uint256 shares;
+        if (assets > 0) {
+            (, shares) = IMorpho(morpho).supply(marketParams(), assets, 0, address(this), hex"");
```

Comment:
`shares` alone (as in [bundler3](https://github.com/morpho-org/bundler3/blob/main/src/adapters/GeneralAdapter1.sol#L89)) is ok, but here there is also the `supplyShares` variable so it's less clear.


```suggestion
            (, mintedShares) = IMorpho(morpho).supply(marketParams(), assets, 0, address(this), hex"");
```

Reply:
PR https://github.com/morpho-org/vault-v2/pull/801